rmw_fastrtps_shared_cpp/src/participant.cpp

```diff
@@ -195,6 +198,26 @@ rmw_fastrtps_shared_cpp::create_participant(
   if (env_value != nullptr) {
     leave_middleware_default_qos = strcmp(env_value, "1") == 0;
   }
+  if (!leave_middleware_default_qos) {
+    error_str = rcutils_get_env("RMW_FASTRTPS_PUBLICATION_MODE", &env_value);
+    if (error_str != NULL) {
+      RMW_SET_ERROR_MSG_WITH_FORMAT_STRING("Error getting env var: %s\n", error_str);
+      return nullptr;
+    }
+    if (env_value != nullptr) {
+      // Synchronous publishing
+      if (strcmp(env_value, "SYNCHRONOUS") == 0) {
+        publishing_mode = publishing_mode_t::SYNCHRONOUS;
+      } else if (strcmp(env_value, "AUTO") == 0) {
+        publishing_mode = publishing_mode_t::AUTO;
+      } else if (strcmp(env_value, "ASYNCHRONOUS") != 0 && strcmp(env_value, "") != 0) {
```

Comment:
```suggestion
      } else if (strcmp(env_value, "ASYNCHRONOUS") != 0) {
```

I think empty case `""` is not admissible based on the design.

Reply:
The environment variable is read using `rcutils_get_env()`:

```
const char *
rcutils_get_env(const char * env_name, const char ** env_value)
{
  RCUTILS_CAN_RETURN_WITH_ERROR_OF("some string error");

  if (NULL == env_name) {
    return "argument env_name is null";
  }
  if (NULL == env_value) {
    return "argument env_value is null";
  }

  // TODO(Suyash458): getenv is deprecated on Windows; consider using getenv_s instead
  *env_value = getenv(env_name);

  if (NULL == *env_value) {
    *env_value = "";
  }
  return NULL;
}
```
It uses function `getenv` that returns `NULL` if the variable does not exist. Below, if `NULL` has been returned, `rcutils_get_env()` returns empty string `""`. Therefore, to avoid the warning log appearance when the environment variable is not defined, this condition should also be checked.

Reply:
that makes sense, appreciate for the explanation.